### ISSUE TITLE
unload reload fails

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#3949]
Trying to test unload/reload in 2.2.2.3706 (a quite new SC build) to prepare for moving to 2.3.x.
But there is so many problems so I don't know where to begin...

Easiest would of course be if the command prompt would work. That just give me
```
Value can not be null.
Parameter name: value
```
Nothing more at all. I get this result no matter what I try for (correctly given) parameters to `staradmin unload`.

So I went with the `Starcounter.Db.Unload(filePath, 0, false);` from an app instead. That seems to work at first, getting a lot of warnings about missing classes but that is according to plan, just good to get rid of some old thrash in the db. However, it still gives an exception after a while:
```
ScErrSchemaCodeMismatch (SCERR4177): Operation failed be because input does not match schema. CLR Class is not loaded for table Concepts.Ring8.Tunity.UsedPerson
Version: 2.2.2.3706
``` 
The thing is that I already have got the correct warning:
```
Table Concepts.Ring8.Tunity.UsedPerson cannot be unloaded, since its class is not loaded.
```

Still this result in a sql, but that sql can not be reloaded. However, It might be better to leave that until I get unload to work correctly.

**Comment by malx122** [comment]
No one?

**Comment by miyconst** [comment]
@malx122 it looks like you are trying to run the `unload` command while not all the CLR classes are loaded. 

If this is the case, please try to start all the apps which define any database classes and call reload once again.

If that is not the case, please share database (if it is not confidential) and exact command line you try to execute.

cc @per-samuelsson, @ubbeK 

P.S. There is a complete new unload/reload tool for Starcounter 2.3.x under development.

**Comment by malx122** [comment]
Yes, of course. The DB is old, there is no way of loading all CLR classes. But thats why you have the `--allowPartial`, I have tried with that one with the same result? And if you see my internal `Starcounter.Db.Unload(filePath, 0, false);`, the last "false" there is the same thing. And the warning says that it shouldn't try but it still tries and fatal fails?

**Comment by malx122** [comment]
bump

**Comment by miyconst** [comment]
@k-rus, @bigwad?

**Comment by k-rus** [comment]
> Easiest would of course be if the command prompt would work. That just give me
> 
> Value can not be null.
> Parameter name: value
> Nothing more at all. I get this result no matter what I try for (correctly given) parameters to staradmin unload.

This one can be looked by @per-samuelsson I guess. But it will be nice to have more detailed description. E.g., copy-paste actual code from command-prompt (not screen shot) and describe what was done before.

**Comment by k-rus** [comment]
> So I went with the Starcounter.Db.Unload(filePath, 0, false); from an app instead. That seems to work at first, getting a lot of warnings about missing classes but that is according to plan, just good to get rid of some old thrash in the db. However, it still gives an exception after a while:
> 
> ScErrSchemaCodeMismatch (SCERR4177): Operation failed be because input does not match schema. CLR Class is not loaded for table Concepts.Ring8.Tunity.UsedPerson
> Version: 2.2.2.3706

Was there any stack trace?

**Comment by per-samuelsson** [comment]
> This one can be looked by @per-samuelsson I guess.

As I see it, the heart of the problem here is that the unload fail with

> ScErrSchemaCodeMismatch (SCERR4177): Operation failed be because input does not match schema. CLR Class is not loaded for table Concepts.Ring8.Tunity.UsedPerson
Version: 2.2.2.3706

even though @malx122 is making a partial unload. I would have no idea as to why that occur.

**Comment by k-rus** [comment]
> This one can be looked by @per-samuelsson I guess.
> As I see it, the heart of the problem here is that the unload fail with

I guess then staradmin should return actual error. But it is low prio and we move away from this tool.

**Comment by per-samuelsson** [comment]
> I guess then staradmin should return actual error.

I was referring to this, where `staradmin` is not at all involved:

> So I went with the `Starcounter.Db.Unload(filePath, 0, false);` from an app instead. That seems to work at first, getting a lot of warnings about missing classes but that is according to plan, just good to get rid of some old thrash in the db. However, it still gives an exception after a while:
>```
>ScErrSchemaCodeMismatch (SCERR4177): Operation failed be because input does not match schema. CLR Class is not loaded for table Concepts.Ring8.Tunity.UsedPerson
>Version: 2.2.2.3706
>``` 
>The thing is that I already have got the correct warning:
>```
>Table Concepts.Ring8.Tunity.UsedPerson cannot be unloaded, since its class is not loaded.
>```

**Comment by k-rus** [comment]
> I guess then staradmin should return actual error.
> I was referring to this, where staradmin is not at all involved:

You were refering to my comment, which was about @malx122 report only:



> Easiest would of course be if the command prompt would work. That just give me
> 
> Value can not be null.
> Parameter name: value
> Nothing more at all. I get this result no matter what I try for (correctly given) parameters to staradmin unload.

I only pointed @per-samuelsson to the above comment on using our command line tool for unload, which fails in this case with meaningless error.

**Comment by malx122** [comment]
Ok, so what do I do? I really like the external "staradmin unload" to work since it make it possible to use my external tool for backup/transfer/copy purposes but at least I need the internal function working correctly.

**Comment by per-samuelsson** [comment]
> Ok, so what do I do? I really like the external "staradmin unload" to work since it make it possible to use my external tool for backup/transfer/copy purposes but at least I need the internal function working correctly.

@malx122, I'll split this up into two issues. One dealing with the poor output of `staradmin`. I can take that. And then I'll update the title of this to pinpoint the error that is that "allow partial unload" has a flaw.
